src/Agent.Worker/TaskManager.cs

```diff
@@ -219,6 +219,7 @@ public sealed class ExecutionData
         private readonly List<HandlerData> _all = new List<HandlerData>();
         private AzurePowerShellHandlerData _azurePowerShell;
         private NodeHandlerData _node;
+        private Node10HandlerData _node10;
```

Comment:
should we make the task.json looks like
```json
"Node": {
            "target": "Some Node10 target",
            "extraNodeArg": "Extra node10 arg value",
            "version": "10"
        },
```

Reply:
This is certainly an option. I followed what the PowerShell/PowerShell3 handlers did as a syntax to keep things consistent, but personally I don't have any strong opinions to not implement the version select as a property.

That said, I'd predict the separate Node/Node10 option to be less "surprising" for the end user (i.e. "why does the agent choose the right PS version with a handler declaration and the Node version with a property?").